modules/ROOT/pages/6.3-release-notes.adoc

```diff
@@ -0,0 +1,694 @@
+
+=== `setContent` is now allowed to accept any custom keys and values as a second options argument
+
+In {productname} 6.3, changes have been made to the `setContent` feature that now allows the editor to accept any custom `[key: string]: (value) any;` as a second option argument when the user inserts content.
+
+The provides more flexibility with the `getContent` and `setContent` features.
+
+New tests have been added to the `setContent` feature in {productname} 6.3 that now check if content manipulation in `BeforeSetContent` callback is possible so it can pass the custom data object to the `getContent` feature, allowing it to set both the content and action events.
+
+
+[[bug-fixes]]
+== Bug fixes
+
+{productname} 6.3 also includes the following bug fixes:
+
+=== Parsing media content would cause a memory leak, which for example occurred when using the getContent API
+
+A memory leak occurred in Chrome, Edge and Firefox browsers when media content was parsed into the editor using the `tinymce.activeEditor.getContent()` method API.
+
+The memory leak caused an abnormal behavior that increased the memory requirements of the editor until the active tab in the browser crashed.
+
+This memory leak has been reported as https://bugs.chromium.org/p/chromium/issues/detail?id=1374180[Memory leak with DOMParser parseFromString method] in the Chromium bug tracker.
+
+A fix that prevents {productname} being affected by the leak has been included in {productname} 6.3.
+
+=== Dragging a noneditable element toward the bottom edge would cause the page to scroll up
+
+A regression was found in {productname} 6.2 where dragging an element containing a `contentEditable=false` attribute towards a document’s edge did not cause scrolling.
+
+It was also found that dragging elements containing a `contentEditable=false` attribute toward the bottom edge causes similar behavior. In this instance, however, when the elements containing a `contentEditable=false` attribute was dragged to the bottom of the page the editor would sometimes scroll up.
+
+This regression has been addressed in {productname} 6.3. Dragging elements containing a `contentEditable=false` attribute towards a document’s edge causes the viewport to scroll as expected. And dragging elements containing a `contentEditable=false` attribute toward the bottom edge of the editor no longer causes the page to scroll up.
+
+=== Range expanding capabilities behaved inconsistently depending on where the insertion point was placed
+
+When the `ExpandRange.ts:expandRng()` function is given a range, it is expected to select the next word whether the range begins before or after the word. Previously, however, this was discovered to not be the case.
+
+As a result of the discovery, it was further discovered that the string `ExpandRange.ts:expandRng()` function, that checks the location of the user’s insertion point before and after, was affecting the for-loop in `walkText-implementation`.
```

Comment:
Don't think this line is relevant as it isn't user-focused.

Reply:
Removed line as not relevant since it is not user-focused.